File: dev/test_data/main/cpp/expected/implementation_specific/input/snippets/verification/texts_are_unique.cpp
bool TextsAreUnique(
  const std::vector<std::wstring>& texts
) {
  std::set<std::wstring> text_set;

  for (const std::wstring& text : texts) {
    if (text_set.find(text) != text_set.end()) {
      return false;
    }

    text_set.insert(text);
  }

  return true;
}

File: dev/test_data/main/java/expected/implementation_specific/input/snippets/Verification/texts_are_unique.java
/**
* Check that the texts do not repeat.
* @param texts the texts to be checked
*/
public static boolean textsAreUnique(Iterable<String> texts) {
  Objects.requireNonNull(texts);

  Set<String> textSet = new HashSet<>();
  for (String text : texts) {
    if (textSet.contains(text)) {
      return false;
    }
    textSet.add(text);
  }
  return true;
}
